probe/docker/reporter.go

```diff
@@ -0,0 +1,66 @@
+package docker
+
+import (
+	docker_client "github.com/fsouza/go-dockerclient"
+
+	"github.com/weaveworks/scope/report"
+)
+
+// Keys for use in NodeMetadata
+const (
+	ContainerName = "docker_container_name"
+	ImageID       = "docker_image_id"
+	ImageName     = "docker_image_name"
+)
+
+// Reporter generate Reports containing Container and ContainerImage topologies
+type Reporter struct {
+	registry Registry
+	scope    string
+}
+
+// NewReporter makes a new Reporter
+func NewReporter(registry Registry, scope string) *Reporter {
+	return &Reporter{
+		registry: registry,
+		scope:    scope,
+	}
+}
+
+// Report generates a Report containing Container and ContainerImage topologies
+func (r *Reporter) Report() report.Report {
+	result := report.MakeReport()
```

Comment:
That would be nice, but I'll leave that up to @peterbourgon and a future change.  Peter is a stickler for naming.